First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains the initial functionality for this Ansible role.

## 💭 Motivation and context ##

We want to have a Linux instance share the EFS share via Samba so that Windows instances can mount it.  A first step toward that is installing Samba on a Linux instance.

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Update metadata for this role

jsf9k committed:
Update README for this role

jsf9k committed:
Use the systemd-enabled molecule configuration

jsf9k committed:
Add the foo that waits for Fedora's systemd to initialize

jsf9k committed:
Create server scenario

jsf9k committed:
Whittle codeowners down to only those folks who know about this repo

jsf9k committed:
Add namespace to metadata

jsf9k committed:
Add code to install packages

jsf9k committed:
Add code to enable Samba service

jsf9k committed:
Add client scenario

jsf9k committed:
Remove unnecessary Dockerfile

jsf9k committed:
Add ability to create Samba guest user

jsf9k committed:
Make the Samba guest user a system user

jsf9k committed:
Add code to configues samba to use the guest user account for guest access

felddy commented on .github/CODEOWNERS:
```diff
@@ -3,7 +3,7 @@
 # These owners will be the default owners for everything in the
 # repo. Unless a later match takes precedence, these owners will be
 # requested for review when someone opens a pull request.
-* @dav3r @felddy @jsf9k @mcdonnnj
```
This saddens me.

jsf9k replied:
I have been doing it for at least a year now.  Given your rare appearances on PRs I thought it made sense.  I am sorry you are sad.

felddy commented on molecule/server/tests/test_server.py:
```diff
@@ -0,0 +1,63 @@
+    """Test that Samba guest user was created."""
+    user = host.user("smbguest")
+    assert user.exists
+    assert user.home == "/nonexistent"
```
This is new to me.  Interesting.

felddy commented on tasks/main.yml:
```diff
@@ -1,2 +1,44 @@
+          ansible.builtin.user:
+            comment: Samba guest account
+            create_home: no
+            home: /nonexistent
```
Not to pick nits, but since @mcdonnnj isn't here... 

Should `home` vary based on the distro?

See: https://unix.stackexchange.com/a/75329

jsf9k replied:
I think as long as it points to a place that doesn't really exist it's fine.  Would you prefer `/dev/null`?

felddy replied:
I would prefer `/dev/null`.  That is what I've always used.  It seems the world has changed.  As long as this isn't a new "bad practice" I think it makes more sense to point to something that is guaranteed not to give a user access.  I could see a configuration like this being part of a hacking challenge.  

I feel heard.  Please feel free to resolve this conversation pending any action or inaction you see fit.

jsf9k replied:
Please see commit 935268e.

felddy reviewed (approved):
Strong work. 💪 

See my one question.

dav3r commented on defaults/main.yml:
```diff
@@ -0,0 +1,11 @@
+# is true.
+create_guest_user: false
+
+# The name of the samba guest user to be created
```
Consistent capitalization:
```suggestion
# The name of the Samba guest user to be created
```

dav3r commented on tasks/main.yml:
```diff
@@ -1,2 +1,44 @@
+            name: "{{ guest_user }}"
+            shell: /sbin/nologin
+            system: yes
+        - name: Configure samba to use the guest account
```
Consistent capitalization:
```suggestion
        - name: Configure Samba to use the guest account
```

dav3r reviewed (approved):
This looks great to me! 🚁

jsf9k committed:
Consistently capitalize "Samba"

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Change home directory to /dev/null

@felddy pointed out that this is a bit more traditional than /nonexistent.

Co-authored-by: Mark Feldhousen <felddy@users.noreply.github.com>

jsf9k committed:
Correct comment

jsf9k committed:
Add package for Samba clients

jsf9k committed:
Add missing word to task name

jsf9k committed:
Install Samba CLI client

mcdonnnj commented on molecule/client/tests/test_client.py:
```diff
@@ -0,0 +1,33 @@
+        host.system_info.distribution == "debian"
+        or host.system_info.distribution == "kali"
+        or host.system_info.distribution == "ubuntu"
+    ):
```
```suggestion
    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
```

mcdonnnj commented on molecule/client/tests/test_client.py:
```diff
@@ -0,0 +1,33 @@
+    elif (
+        host.system_info.distribution == "fedora"
+        or host.system_info.distribution == "amzn"
+    ):
```
```suggestion
    elif host.system_info.distribution in ["amzn", "fedora"]:
```

mcdonnnj commented on molecule/server/tests/test_server.py:
```diff
@@ -0,0 +1,63 @@
+        host.system_info.distribution == "debian"
+        or host.system_info.distribution == "kali"
+        or host.system_info.distribution == "ubuntu"
+    ):
```
```suggestion
    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
```

mcdonnnj commented on molecule/server/tests/test_server.py:
```diff
@@ -0,0 +1,63 @@
+    elif (
+        host.system_info.distribution == "fedora"
+        or host.system_info.distribution == "amzn"
+    ):
```
```suggestion
    elif host.system_info.distribution in ["amzn", "fedora"]:
```

mcdonnnj commented on molecule/server/tests/test_server.py:
```diff
@@ -0,0 +1,63 @@
+        host.system_info.distribution == "debian"
+        or host.system_info.distribution == "kali"
+        or host.system_info.distribution == "ubuntu"
+    ):
```
```suggestion
    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
```

mcdonnnj commented on molecule/server/tests/test_server.py:
```diff
@@ -0,0 +1,63 @@
+    elif (
+        host.system_info.distribution == "fedora"
+        or host.system_info.distribution == "amzn"
+    ):
```
```suggestion
    elif host.system_info.distribution in ["amzn", "fedora"]:
```

mcdonnnj commented on molecule/server/prepare.yml:
```diff
@@ -0,0 +1,34 @@
+- name: Wait for systemd to complete initialization (Fedora)
+  hosts: os_Fedora
+  tasks:
+    - name: Wait for systemd to complete initialization # noqa 303
```
What are we `noqa`ing here and why?

jsf9k replied:
Please see commit 3185672.

mcdonnnj replied:
Perfect, thanks!

mcdonnnj commented on README.md:
```diff
@@ -4,28 +4,21 @@
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for an Ansible role.
+This is an Ansible role that installs the dependencies necessary for
+Samba.
```
```suggestion
[Samba](https://www.samba.org/).
```

mcdonnnj reviewed (changes requested):
Nice and clean! Mostly just some suggestions you can take or leave, but I did have a question about a `noqa` I would like answered.

jsf9k committed:
Add link to Samba web page

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Prefer checking if a string is in an array vs an if clause with a bunch of ors

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add a comment explaining why a noqa for ansible-lint is present

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add two optional role variables

If present, these variables specify the Samba guest user's UID and
additional groups.

Also do not require the Samba guest user to be a system user.  We want
to allow it to have the same UID as the vncuser that is used
elsewhere.
